pacman_mirrors/pacman_mirrors.py

```diff
@@ -165,64 +154,131 @@ def command_line_parse(self):
+            self.output_mirror_list(server_list, write_file=True)
+            if self.custom:
+                self.modify_config(custom=True)
+            else:
+                self.modify_config()
+        else:
+            print("\n.: {}: {}\n".format(txt.WRN_CLR, txt.ERR_NO_MIRRORS))
+
+    def gen_mirror_list_interactive(self):
+        """Prompt the user to select the mirrors with a gui.
+        * Outputs a pacman mirrorlist,
+        * Outputs a "custom" mirror file
+        * Modify the configuration file to use the "custom" file.
+        """
+        random = self.config["method"] == "random"
+        mirrorlist = []
+        interactive_list = []
+        for country in self.only_country:
+            for mirror in self.mirrors.mirrorlist:
+                if country == mirror["country"]:
+                    mirrorlist.append(mirror)
+                    interactive_list.append({
+                        "country": mirror["country"],
+                        "resp_time": mirror["resp_time"],
+                        "last_sync": mirror["last_sync"],
+                        "url": mirror["url"]
+                    })
+                    interactive_list = sorted(interactive_list, key=itemgetter("resp_time"))
+                    if random:
+                        shuffle(interactive_list)
+        if self.no_display:
+            from . import consoleui as ui
+        else:
+            from . import graphical_ui as ui
+        interactive = ui.run(interactive_list, random)
+        if interactive.is_done:
+            selected = []
+            mirrorfile = []
+            for item in interactive.custom_list:
+                for server in self.mirrors.mirrorlist:
+                    if item["url"] == server["url"]:
+                        selected.append(server)
+                        mirrorfile.append({
+                            "country": server["country"],
+                            "protocols": server["protocols"],
+                            "url": server["url"]
+                        })
+            if mirrorfile:
+                print("\n.: {}: {}".format(txt.INF_CLR, txt.INF_INTERACTIVE_LIST))
+                print("--------------------------")
+                # output mirror file
+                JsonFn.write_json_file(mirrorfile, CUSTOM_FILE)
+                print(".: {} {}: {}".format(txt.INF_CLR, txt.INF_MIRROR_FILE_SAVED, CUSTOM_FILE))
+                # output pacman mirrorlist
+                self.output_mirror_list(selected, write_file=True)
+                # output custom configuration
+                self.config["only_country"] = ["Custom"]
+                self.modify_config(custom=True)
+                print(".: {} {}: {}".format(txt.INF_CLR, txt.INF_INTERACTIVE_LIST_SAVED, CUSTOM_FILE))
+            else:
+                print(".: {} {}".format(txt.WRN_CLR, txt.INF_NO_SELECTION))
+                print(".: {} {}".format(txt.WRN_CLR, txt.INF_NO_CHANGES))
+        else:
```

Comment:
I humbly disagree.
The only way a user can exit from TUI is by clicking OK - so in that case it is nice to know nothing has changed.